Client/Frontend/Browser/SearchViewController.swift

```diff
@@ -213,48 +242,34 @@ class SearchViewController: SiteTableViewController, KeyboardHelperDelegate {
 
 extension SearchViewController: UITableViewDataSource {
     override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
-        let cell = super.tableView(tableView, cellForRowAtIndexPath: indexPath)
-
-        if let currentSection = SearchListSection(rawValue: indexPath.section) {
-            switch currentSection {
-            case .SearchSuggestions:
-                cell.textLabel?.text = searchSuggestions[indexPath.row]
-                cell.detailTextLabel?.text = nil
-                cell.imageView?.image = nil
-                cell.isAccessibilityElement = false
-                cell.accessibilityLabel = nil
-            case .BookmarksAndHistory:
-                if let site = data[indexPath.row] as? Site {
-                    cell.textLabel?.text = site.title
-                    cell.detailTextLabel?.text = site.url
-                    if let img = site.icon? {
-                        let imgUrl = NSURL(string: img.url)
-                        cell.imageView?.sd_setImageWithURL(imgUrl, placeholderImage: self.profile.favicons.defaultIcon)
-                    } else {
-                        cell.imageView?.image = self.profile.favicons.defaultIcon
-                    }
-                    cell.isAccessibilityElement = false
-                    cell.accessibilityLabel = nil
+        switch SearchListSection(rawValue: indexPath.section)! {
```

Comment:
There's a tradeoff here between developer time and user experience. Compare crashes to things like history results intermittently not appearing, or favicons occasionally showing the wrong image. It can take a lot of effort to find reliable STR, and then a lot more effort tracking down the cause in the code.

Also consider how often we hit NPEs (or any other exception) while we're developing (a lot!). Now imagine finding and tracking down each of those instances with no stack trace, but instead glitchy app behavior. Now all of those NPEs become our friend.

> Swift gives us tools that force us to write code that is correct. If you use those features then there cannot be a case of "where we forgot to assign them" - simply because that is not possible.

Not crashing != correct. And I don't know what you mean in your second sentence -- of course we can still forget to assign things, whether we check them or not. And if something _requires_ something be set to function properly as a precondition, I might as well enforce it, no?

For this case in particular, consider what could cause this to break. We have an enum with two specific int values, and we're converting back to that enum from ints from that enum. This isn't some obscure code hidden deep in some callback chain. We have a very clear understanding of what's going on here, so I'm comfortable strongly asserting that this should never happen.

To be clear, I'm not advocating force-unwrapping optionals everywhere. They should be used judiciously. What are the chances this particular case will be wrong? Very low, and if it is wrong, it's almost certainly because of a newly introduced change as a developer -- and I would want to catch it immediately.